census_view.cpp

```diff
@@ -932,13 +936,815 @@ inline std::vector<std::string> const& CensusViewDataViewModel::all_headers() co
     return view_.case_parms()[0].member_names();
 }
 
-// class CensusView
+namespace
+{
+
+// Declare the functions here, but implement after the CensusViewTable
+// declaration to avoid the "use of undefined type" error.
+
+// Get the cell value from the table.
+
+any_member<Input> const& cell_at(wxGridTableBase const& table, int row, int col);
+
+// Get the raw value from the table.
+
+Input const& row_at(wxGridTableBase const& table, int row);
+
+// Get the column name from the table.
+
+std::string const& col_name(wxGridTableBase const& table, int col);
+
+// class RangeDoubleEditor
+
+class RangeDoubleEditor
+    :public wxGridCellTextEditor
```

Comment:
I wonder if we should create a grid editor using `wxSpinCtrlDouble`. Does DVC use one or does it just use a text as well?

Reply:
DVC version uses wxTextCtrl:
https://github.com/vadz/lmi/blob/master/census_view.cpp#L300